### ISSUE TITLE
[py] replacing seed with random_state in sklearn API

### ISSUE THREAD

**Issue by tlorieul** [dmlc/xgboost#1182]
In sklearn, the seed is usually set using an argument named `random_state` instead of `seed` in `XGBRegressor` and `XGBClassifier`. Thus, if one wants to do bagging for example through `sklearn.ensemble.BaggingClassifier` with `XGBClassifier`, the bagging part (sampling with replacement) would be ok but all the models will have the same seed.

This is not really hard to fix, I can do a PR if this modification is accepted. However, there are some small issues linked to backward compatibility that I would like to discuss first:
- should `seed` simply be deleted and replaced by `random_state` or should it be let as it is with a decrecation warning?
- in sklearn the default argument for the seed is `random_state=None` which in fact sets a seed at random instead to 0, thus the default behavior is not reproducible which also somewhat breaks backward compatibility, is it better in this case to set `random_state=0` or not?

**Comment by tqchen** [comment]
Let us simply keep seed argument, and when it is not available, accept random_state instead

**Comment by tlorieul** [comment]
Sorry, I'm confused: what do you mean exactly by "when it is not available"? Do you mean when it is explicitly set to `None`?

**Comment by tqchen** [comment]
OK, I checked it again. For scikit-learn API. I think it is acceptable to rename seed as random_state, if that is what people typically use. Sorry about confusion in my last reply

**Comment by tlorieul** [comment]
Ok, no worries ;) ! I'll send a PR when I've done the changes.

And concerning the backward compatibility issues, should I put a deprecation warning for `seed` ? For the default value of `random_state` should I let it to 0 as it is currently the case for `seed` or should I set it to `None` as it is the default value for this argument in sklearn which will sample a random seed at each execution?
